### PR TITLE
qa/tasks/cephfs: umount the mountpoints when tearDown

### PR DESCRIPTION
**Pull request by lxbsz**
Fixes: https://tracker.ceph.com/issues/44408

All the tempory mountpoints still kept after the test case finished.

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: qa/tasks/vstart_runner.py [hunk 1]
```diff
@@ -617,31 +617,8 @@ def _asok_path(self):
...
-
-        rproc = self.client_remote.run(args=[
-                'rmdir',
-                '--',
-                self.mountpoint,
-            ])
-        rproc.wait()
-        self.mounted = False
+        if self.is_mounted():
+            super(LocalKernelMount, self).umount()
```
Comment (batrick) [inline qa/tasks/vstart_runner.py]: I think we can just undefine the method so it automatically uses the parent?

File: qa/tasks/cephfs/xfstests_dev.py [hunk 2]
```diff
@@ -171,3 +171,5 @@ def tearDown(self):
         self.mount_a.client_remote.run(args=['sudo', 'rm', '-rf',
                                              self.repo_path],
                                        omit_sudo=False, check_status=False)
+        
+        super(XFSTestsDev, self).tearDown()
```
Comment (batrick) [inline qa/tasks/cephfs/xfstests_dev.py]: cc @rishabh-d-dave

File: qa/tasks/vstart_runner.py (hunk 1 above)
Comment (lxbsz) [reply qa/tasks/vstart_runner.py]: Yeah, will fix it.

File: qa/tasks/cephfs/xfstests_dev.py [hunk 3]
```diff
@@ -172,3 +172,5 @@ def tearDown(self):
         self.mount_a.client_remote.run(args=['sudo', 'rm', '-rf',
                                              self.repo_path],
                                        omit_sudo=False, check_status=False)
+
```
Comment (rishabh-d-dave) [inline qa/tasks/cephfs/xfstests_dev.py]: There's whitespace here, see https://github.com/lxbsz/ceph/blob/qa_umount_failed/qa/tasks/cephfs/xfstests_dev.py#L175.

Comment (lxbsz) [reply qa/tasks/cephfs/xfstests_dev.py]: Good catch and fixed it.
Thanks.

**Comment by gregsfortytwo** [comment]
jenkins test make check

**Comment by gregsfortytwo** [comment]
@lxbsz I thought the tests were mostly responsible for tearing down their own mounts, and if the mount was left behind that indicated an error. Is that wrong?

**Comment by lxbsz** [comment]
> @lxbsz I thought the tests were mostly responsible for tearing down their own mounts, and if the mount was left behind that indicated an error. Is that wrong?

Yeah, that's right.

Here for the cephfs test cases, the test cases' parent class's setUp method will help do the normal mount operation before running the test cases and the Teardown method will do the umount. Then in some test cases, if it need some more extra mount opts for the mount points, they should do the remount.

But currently for the TestACLs(XFSTestsDev) test case, after the TestACLs finished, the mount/umount is unpaired, because the XFSTestsDev Teardown didn't trigger the umount.

**Comment by gregsfortytwo** [comment]
Oh I misunderstood what was happening here, looks good.

**Comment by gregsfortytwo** [comment]
http://pulpito.front.sepia.ceph.com/gregf-2020-04-15_20:54:50-fs-wip-greg-testing-415-1-distro-basic-smithi/ failures are https://tracker.ceph.com/issues/45133, https://tracker.ceph.com/issues/45103, and a couple rpm metadata download errors.

**Pull request merged**